google-cloud-apis/java-apis/maven-apis/src/com/google/cloud/tools/intellij/cloudapis/maven/MavenCloudApiUiExtension.java

```diff
@@ -16,44 +16,77 @@
 
 package com.google.cloud.tools.intellij.cloudapis.maven;
 
+import static com.google.cloud.tools.intellij.cloudapis.GoogleCloudApiDetailsPanel.makeLink;
+
 import com.google.cloud.tools.intellij.GoogleCloudCoreIcons;
 import com.google.cloud.tools.intellij.cloudapis.CloudApiUiExtension;
 import com.google.cloud.tools.intellij.cloudapis.CloudApiUiPresenter;
 import com.google.cloud.tools.intellij.cloudapis.maven.CloudApiMavenService.LibraryVersionFromBomException;
 import com.google.cloud.tools.intellij.util.ThreadUtil;
 import com.google.cloud.tools.libraries.json.CloudLibrary;
 import com.google.cloud.tools.libraries.json.CloudLibraryClientMavenCoordinates;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
 import com.intellij.icons.AllIcons.General;
 import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.application.ModalityState;
 import com.intellij.openapi.module.Module;
-import java.util.Collection;
-import java.util.Collections;
+import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import javax.swing.JComponent;
+import javax.swing.JLabel;
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * {@link CloudApiUiExtension} to support additional components and events for Maven Cloud API
  * integration.
  */
 public class MavenCloudApiUiExtension implements CloudApiUiExtension {
+  private BomComboBox bomComboBox;
+  private JLabel bomSelectorLabel;
+
+  private CloudLibrary currentCloudLibrary;
 
   @Override
-  public Collection<JComponent> createCustomUiComponents() {
-    return Collections.emptyList();
+  public Map<EXTENSION_UI_COMPONENT_LOCATION, JComponent> createCustomUiComponents() {
```

Comment:
Perhaps include a javadoc here (or at the class level) to indicate exactly what this is adding in the UI so that its clear from a quick browse of this class.